### PR TITLE
TIMG clusters

### PR DESCRIPTION
**Pull request by burrbull**
~~Depends on https://github.com/esp-rs/esp-pacs/pull/220~~

### CODE REVIEW

File: esp-hal/src/timer.rs [hunk 1]
```diff
@@ -156,7 +157,8 @@ impl TimerGroupInstance for TIMG1 {
     fn configure_src_clk() {
         unsafe {
             (*Self::register_block())
-                .t1config()
```
Comment (burrbull) [inline esp-hal/src/timer.rs:159]: I'm not familiar with ESP timers, but is this correct?

Comment (MabezDev) [reply esp-hal/src/timer.rs:159]: Looks okay to me, but I will do some testing on my end too

Comment (MabezDev) [reply esp-hal/src/timer.rs:159]: TIMG0 & TIMG1 still work correctly on a esp32s2 and esp32s3 board I tested here

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/timer.rs (hunk 1 above)
Comment (burrbull) [reply esp-hal/src/timer.rs:159]: should there be t(0).config for both TIMG0 and TIMG1?

Comment (MabezDev) [reply esp-hal/src/timer.rs:159]: image

Huh, yeah I think you're right! It seems each timer in each timer group can select its clock source individually, I will create an issue. Thanks!

**Pull request merged**